core/modules/css/KvInterfaceImplMem.cc

```diff
@@ -257,38 +279,40 @@ KvInterfaceImplMem::deleteKey(string const& key) {
     }
 }
 
-std::string KvInterfaceImplMem::dumpKV() {
-    std::string result;
+std::string KvInterfaceImplMem::dumpKV(std::string const& key) {
+    const string pfx(key == "/" ? key : key + "/");
+    ptree::ptree tree;
     for (auto& pair: _kvMap) {
-        if (not result.empty()) result += '\n';
-        result += pair.first;
-        result += '\t';
-        if (pair.second.empty()) {
-            result += "\\N";
-        } else {
-            if (pair.second.find('\n') != std::string::npos) {
-                throw CssError("KvInterfaceImplMem::dumpKV - value contains newline");
-            }
-            result += pair.second;
+        if (pair.first.empty()) {
+            // skip root key, it is not useful and not supposed to have any data
+            continue;
+        }
```

Comment:
It might be nice to see the root element when looking at the output so you know it is starting with the root element.

Reply:
Our data model for CSS/KV does not need root element, it is specified entirely in terms of its leaf paths. I can expect some implementation don't have it all (I suspect that in KvInterfaceImplMem we can drop root element without any change in behavior). So I prefer to keep it invisible in the DUMP output, there was a code in mysql implementation which already done that, I have just extended it to in-memory KVI (and fixed other small inconsistencies).